src/main/java/com/example/database/services/OrderDetailsService.java

```diff
@@ -0,0 +1,112 @@
+package com.example.database.services;
+
+import com.example.business.models.ItemModel;
+import com.example.business.models.OrderDetailsModel;
+import com.example.business.models.ProductModel;
+import com.example.database.entity.Address;
+import com.example.database.entity.ItemQuantity;
+import com.example.database.entity.OrderDetails;
+import com.example.database.entity.UserEntity;
+import com.example.database.repositories.interfaces.AddressRepository;
+import com.example.database.repositories.interfaces.ItemQuantityRepository;
+import com.example.database.repositories.interfaces.OrderDetailsRepository;
+import com.example.database.repositories.interfaces.UserRepository;
+import com.example.utils.converters.OrderDetailsConverter;
+import com.example.utils.converters.ProductConverter;
+import io.smallrye.mutiny.Uni;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.transaction.Transactional;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.smallrye.mutiny.Uni.combine;
+import static java.util.stream.Collectors.toList;
+
+@ApplicationScoped
+public class OrderDetailsService {
+
+    private final OrderDetailsRepository orderDetailsRepository;
+    private final ItemQuantityRepository itemQuantityRepository;
+    private final AddressRepository addressRepository;
+    private final UserRepository userRepository;
+    private final ItemService itemService;
+
+    public OrderDetailsService(OrderDetailsRepository orderDetailsRepository, ItemQuantityRepository itemQuantityRepository, AddressRepository addressRepository, UserRepository userRepository, ItemService itemService) {
```

Comment:
In my previous company, when having longer dependency list, we separated list, to be more readable, like this:
```suggestion
    public OrderDetailsService(
        OrderDetailsRepository orderDetailsRepository, 
        ItemQuantityRepository itemQuantityRepository, 
        AddressRepository addressRepository, 
        UserRepository userRepository, 
        ItemService itemService
    ) {
```